### DISCUSSION TITLE
handleDrop with async functions

### DISCUSSION THREAD

**Discussion by gee4vee** [microsoft/vscode-discussions#2026] [Extension Development QnA] [answered]
Is it possible to call async functions within `handleDrop` or `handleDrag`? I'm getting errors trying to change the signature to support the async modifier.

**Comment by alefragnani** [comment]
Hi @gee4vee ,

It should work just fine, as both functions are async (`Thenable`).

Take a look at this sample (https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/testViewDragAndDrop.ts) in the vscode-extensions-repo, as reference.

Hope this helps

**Reply by gee4vee** [reply, marked as answer]
Actually looks like I dont get an error anymore. I was able to add `async` and change to `Promise<void>` for the return type. Thanks!

**Comment by gee4vee** [comment]
@alefragnani So I was following the sample but it has no use of async. If I add the async modifier I get an error forcing me to change the signature to return Promise instead of Thenable, but then that breaks the inheritance. Do you know the correct way to do it?
